cpp/include/raft/distance/detail/distance_ops/l2_exp.cuh

```diff
@@ -102,7 +102,10 @@ struct l2_exp_distance_op {
     }
   }
 
-  l2_exp_cutlass_op<DataT, AccT> get_cutlass_op() { return l2_exp_cutlass_op<DataT, AccT>(sqrt); }
+  constexpr l2_exp_cutlass_op<DataT, AccT> get_cutlass_op()
+  {
+    return l2_exp_cutlass_op<DataT, AccT>(sqrt);
```

Comment:
Is this path being tested? Where's the `sqrt` coming from?

Reply:
Short story:
- This path is tested in `cpp/test/distance/dist_l2_sqrt_exp.cu`
- the sqrt is coming from the l2_expanded_distance op. It is an instance variable.

Long story:
The `DistanceType::L2SqrtExpanded` and `DistanceType::L2Expanded` distance types both map to an l2_exp_distance_op. The l2 expanded distance operation has an instance boolean variable `sqrt` which determines whether to calculate the square root. The distance op (and its member variable `sqrt`) gets passed to the kernel at runtime and the kernel decides at runtime whether to compute the square root or not.